src/sdks/core/src/cpp/sdk/cpptest/unit/deviceTest.cpp

```diff
@@ -0,0 +1,268 @@
+    EXPECT_EQ(model, REMOVE_QUOTES(actual_value));
+}
+
+TEST_F(DeviceTest, Sku)
+{
+
+    auto actual_value = jsonEngine->get_value("Device.sku");
+
+    auto sku = Firebolt::IFireboltAccessor::Instance().DeviceInterface().sku(&error);
+
+    EXPECT_EQ(error, Firebolt::Error::None) << "Failed to retrieve sku from Device.sku() method";
+    EXPECT_EQ(sku, REMOVE_QUOTES(actual_value));
+}
+
+TEST_F(DeviceTest, TestDeviceMake)
+{
+
+    auto actual_value = jsonEngine->get_value("Device.make");
+
+    auto make = Firebolt::IFireboltAccessor::Instance().DeviceInterface().make(&error);
+
+    EXPECT_EQ(error, Firebolt::Error::None) << "Failed to retrieve make from Device.make() method";
+    EXPECT_EQ(make, REMOVE_QUOTES(actual_value));
+}
+
+TEST_F(DeviceTest, Hdcp)
+{
+    Firebolt::Error error = Firebolt::Error::None;
+
+    // Parsing expected JSON values into a BooleanMap
+    nlohmann::json expectedJson = nlohmann::json::parse(jsonEngine->get_value("Device.hdcp"));
+    Firebolt::Types::BooleanMap expectedValues;
+
+    for (auto it = expectedJson.begin(); it != expectedJson.end(); ++it)
+    {
+        expectedValues[it.key()] = it.value().get<bool>();
+    }
+
+    // Getting the actual value from the DeviceInterface
+    Firebolt::Device::HDCPVersionMap hdcpMap = Firebolt::IFireboltAccessor::Instance().DeviceInterface().hdcp(&error);
+
+    EXPECT_EQ(error, Firebolt::Error::None) << "Failed to retrieve hdcp from Device.hdcp() method";
+
+    // Convert HDCPVersionMap to BooleanMap for comparison
+    Firebolt::Types::BooleanMap actualValues;
+    actualValues["hdcp1.4"] = hdcpMap.hdcp1_4;
+    actualValues["hdcp2.2"] = hdcpMap.hdcp2_2;
+
+    EXPECT_EQ(actualValues, expectedValues);
+}
+
+TEST_F(DeviceTest, Hdr)
+{
+    Firebolt::Error error = Firebolt::Error::None;
+
+    // Parsing expected JSON values into a BooleanMap
+    nlohmann::json expectedJson = nlohmann::json::parse(jsonEngine->get_value("Device.hdr"));
+    Firebolt::Types::BooleanMap expectedValues;
+
+    for (auto it = expectedJson.begin(); it != expectedJson.end(); ++it)
+    {
+        expectedValues[it.key()] = it.value().get<bool>();
+    }
+
+    // Getting the actual value from the DeviceInterface
+    Firebolt::Device::HDRFormatMap hdrMap = Firebolt::IFireboltAccessor::Instance().DeviceInterface().hdr(&error);
+
+    EXPECT_EQ(error, Firebolt::Error::None) << "Failed to retrieve hdr from Device.hdr() method";
+
+    // Convert HDRFormatMap to BooleanMap for comparison
+    Firebolt::Types::BooleanMap actualValues;
+    actualValues["hdr10"] = hdrMap.hdr10;
+    actualValues["hdr10Plus"] = hdrMap.hdr10Plus;
+    actualValues["dolbyVision"] = hdrMap.dolbyVision;
+    actualValues["hlg"] = hdrMap.hlg;
+
+    EXPECT_EQ(actualValues, expectedValues);
+}
+
+TEST_F(DeviceTest, Audio)
+{ 
+    // Hardcoded expected values
+    Firebolt::Device::AudioProfiles expectedValues;
+    expectedValues.stereo = true;
```

Comment:
Why are we hard-coding the values instead of getting them from the OpenRPC here? I'm assuming it has something to do with the name but if those values ever change this will fail.